speech-to-speech/README.md

```diff
@@ -0,0 +1,88 @@
+- [Cocoapods][cocoapods] version 1.0 or later
+
+## Setup
+
+- Create a project (or use an existing one) in the [Google Cloud Console][cloud-console]
+- [Enable billing][billing] and the
+    - [Speech API](https://console.cloud.google.com/apis/library/speech.googleapis.com).
+    - [Translate API](https://console.cloud.google.com/apis/library/translate.googleapis.com).
+    - [Text-to-Speech API](https://console.cloud.google.com/apis/library/texttospeech.googleapis.com).
+    - [IAM Service Account Credentials API](https://console.cloud.google.com/apis/library/iamcredentials.googleapis.com).
+- [Create a Service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the following IAM role: `Cloud Translation API User`. Example name: `translate-client`. ([For more info on: how to add roles to a Service Account](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource))
+
+###  Setup the app
+- Clone this repository `git clone https://github.com/GoogleCloudPlatform/ios-docs-samples.git` 
+- `cd ios-docs-samples/speech-to-speech/` 
+- Run `./INSTALL-COCOAPODS` to install app dependencies (this can take few minutes to run). When it finishes, it will open the SpeechtoSpeech workspace in Xcode. Since we are using Cocoapods, be sure to open the `SpeechtoSpeech.xcworkspace` and not `SpeechtoSpeech.xcodeproj`.
+- In Xcode's Project Navigator, open the `ApplicationConstants.swift` file within the `SpeechtoSpeech` directory.
+- Find the line where the `project-id` is set. Replace the string value with the identifier of your Google Cloud project.
+
+###  Setup Firebase on the application:
+
+- Complete the steps for [Add Firebase to your app](https://firebase.google.com/docs/ios/setup#add_firebase_to_your_app) and expand the "Create a Firebase project" section for instructions on how to add project to your Firebase console. Note: No need to complete any other sections, they are already done. 
+- Use `iOS bundle ID` as `com.sample.SpeechtoSpeech`
+- In the [Firebase console](https://console.firebase.google.com/), open the "Authentication" section under Develop.
+- On the **Sign-in Methods** page, enable the **Anonymous** sign-in method.
+
+###  Setup and Deploy the Firebase Function 
+
+The Firebase Function provides auth tokens to your app, You'll be using a provided sample function to be run with this app.
+
+- Follow the steps in this [guide](https://firebase.google.com/docs/functions/get-started) for: 
+- "1. Set up Node.js and the Firebase CLI"
+- "2. Initialize Firebase SDK for Cloud Functions".
+- Replace `index.js` file with the [provided index.js](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/functions/dialogflow/functions/index.js).
+- Replace scope in line 79 with `scope: ['https://www.googleapis.com/auth/cloud-platform'],`
+- Replace `DialogflowTokens` with `SpeechTokens` in Line number 30 and 152.
```

Comment:
Maybe we should push an update to the Function and Dialogflow tutorials to some of these variables more generic so that people don't have to edit these lines each time.

Reply:
Yes, Can we update the functions code like below?

`const tokenCollection = 'DialogflowTokens' (or) 'SpeechTokens';
const token = 'OauthToken'`;